docs/DeveloperGuide.md

```diff
@@ -36,3 +92,351 @@
+    MEDIUM,
+    HARD
+}
+```
+
+### Alternatives Considered
+
+An alternative to using an enum could be to use integer constants or string constants to represent the difficulty levels. However, enums provide type safety and better code readability, making them a preferred choice for representing a fixed set of values.
+
+## Class Diagrams
+
+```
++-------------+
+|DifficultyLevel|
++-------------+
+| EASY        |
+| MEDIUM      |
+| HARD        |
++-------------+
+
++------------+
+|     Ai     |
++------------+
+| -minDirection: int
+| -maxDirection: int
++------------+
+| +Ai(difficultyLevel: DifficultyLevel)
+| +getAiDirection(): int
+| +getMinDirection(): int
+| +getMaxDirection(): int
++------------+
+
++----------------+
+|     Penalty    |
++----------------+
+| -PENALTY_PROMPT: String
++----------------+
+| +executePenalty(ai: Ai): void
+| -getUserPenaltyDirection(): int
+| +convertToShootDirection(direction: String): int
++----------------+
+
++----------------+
+|   CommandList  |
++----------------+
+| +goalCheck(userInput: int, aiInput: int): boolean
++----------------+
+
++-------------+
+|  Formatter  |
++-------------+
+| +printGoalAfterShot(goalScored: boolean): void
+| +printErrorWrongArgumentType(commandName: String, regex: String, argumentPosition: int): void
++-------------+
+
++--------+
+|   Ui   |
++--------+
+| -userInput: String
++--------+
+
+```
+
+The `Ai` class depends on the `DifficultyLevel` enum to determine the range of directions for the AI. The `Penalty` class uses the `Ai` class to get the AI's direction and compares it with the user's input direction. The `CommandList` class provides the `goalCheck()` method to determine if a goal is scored based on the user's and AI's directions. The `Formatter` class is responsible for printing the goal result and error messages related to wrong argument types. The `Ui` class provides the user's input direction to the `Penalty` class.
+
+## Future Enhancements
+
+1. **Adaptive AI Difficulty**: Implement an adaptive AI difficulty system where the AI's behavior changes based on the player's performance. This could involve adjusting the range of directions or introducing other challenges for the AI.
+
+2. **Penalty Shootout Mode**: Implement a penalty shootout mode where the game is decided by a series of penalty kicks. This could involve additional rules and scoring mechanisms specific to penalty shootouts.
+
+3. **Custom Difficulty Levels**: Allow users to define custom difficulty levels by specifying the range of directions or other parameters that affect the AI's behavior.
+
+4. **Advanced AI Strategies**: Enhance the AI's decision-making capabilities by implementing more advanced strategies and algorithms. This could involve considering factors such as the player's previous shots, game statistics, or other relevant data.
+
+5. **Multiplayer Mode**: Implement a multiplayer mode where multiple players can participate in the game, either locally or over a network.
```

Comment:
Good job adding future plans and enhancements along with the existing implemented features!